source/infrastructure/bin/aws-instance-scheduler.ts

```diff
@@ -18,48 +18,47 @@ import * as cdk from 'aws-cdk-lib';
 import { AwsInstanceSchedulerStack } from '../lib/aws-instance-scheduler-stack';
 import { AwsInstanceSchedulerRemoteStack } from '../lib/aws-instance-scheduler-remote-stack';
 import PipelineStack from "../pipeline/pipeline-stack";
-import {Aspects} from "aws-cdk-lib";
+import {Aspects, DefaultStackSynthesizer} from "aws-cdk-lib";
 import {AwsSolutionsChecks, NagSuppressions} from "cdk-nag";
+import {getSolutionContext} from "./cdk-context";
 
-const SOLUTION_VERSION = process.env['DIST_VERSION'] || '%%VERSION%%';
-const SOLUTION_NAME = process.env['SOLUTION_NAME'] ? process.env['SOLUTION_NAME'] : "aws-instance-scheduler";
-const SOLUTION_ID = process.env['SOLUTION_ID'] ? process.env['SOLUTION_ID'] : "SO0030";
-const SOLUTION_BUCKET = process.env['DIST_OUTPUT_BUCKET'] ? process.env['DIST_OUTPUT_BUCKET'] : "";
-const SOLUTION_TMN = process.env['SOLUTION_TRADEMARKEDNAME'] ? process.env['SOLUTION_TRADEMARKEDNAME'] : "aws-instance-scheduler";
-const SOLUTION_PROVIDER = 'AWS Solution Development';
-const APP_REG_APPLICATION_TYPE = process.env['APP_REG_APPLICATION_TYPE'] ? process.env['APP_REG_APPLICATION_TYPE'] : 'AWS-Solutions';
-const APP_REG_SOLUTION_NAME = process.env['APP_REG_SOLUTION_NAME'] ? process.env['APP_REG_SOLUTION_NAME'] : "instance-scheduler-on-aws";
+
+let synthesizer = new DefaultStackSynthesizer({
+    generateBootstrapVersionRule: false,
+});
+
+// Solutions pipeline deployment
+const { DIST_OUTPUT_BUCKET, SOLUTION_NAME, VERSION } = process.env;
+if (DIST_OUTPUT_BUCKET && SOLUTION_NAME && VERSION)
+    synthesizer = new DefaultStackSynthesizer({
+        generateBootstrapVersionRule: false,
+        fileAssetsBucketName: `${DIST_OUTPUT_BUCKET}-\${AWS::Region}`,
```

Comment:
is this backslash needed? I can't understand why

Reply:
It's to escape the $ so that CFN can fill in the region. (or so I presume, I copy-pasted this from SIH)